modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/dsl/StateMonitoring.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2014-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.rdbloader.dsl
+
+import java.time.{ Duration, Instant }
+
+import scala.concurrent.duration._
+
+import cats.{ Monad }
+import cats.implicits._
+
+import cats.effect.Timer
+
+import com.snowplowanalytics.snowplow.rdbloader.State
+import com.snowplowanalytics.snowplow.rdbloader.loading.Load
+
+object StateMonitoring {
+
+  val DefaultExtendPeriod = 3.minutes
+
+  def run[F[_]: Monad: Timer: Logging](globalState: State.Ref[F], extend: FiniteDuration => F[Unit]) = {
+    def raise(loading: Load.State, updated: Instant): F[Unit] =
+      Timer[F].clock.instantNow.flatMap { now =>
+        val duration = Duration.between(now, updated).toMinutes
+        Logging[F].warning(show"Loading has stuck. $loading. Spent $duration minutes at this stage. Extending processing for $DefaultExtendPeriod")
```

Comment:
This log message is likely to be scrutinized whenever something is wrong with Redshift, so it is worth spending time getting the language right.  Suggestion:

```
show"Loading is ongoing but approaching deadline. $loading. Spent $duration minutes at this stage. Extending processing deadline for $DefaultExtendPeriod"
```

I didn't like the word "stuck".  The point of your change is to not be stuck.